src/wazuh_modules/syscollector/sysInfoWin.cpp

```diff
@@ -8,18 +8,280 @@
+            {
+                static_cast<int>(GetLastError()),
+                std::system_category(),
+                "Unable to retrieve executable path from process with PID %lu" + m_pId
+            };
+        }
+        return ret;
+    }
+
+    ULONGLONG kernelModeTime() const
+    {
+        return m_kernelModeTime.QuadPart;
+    }
+
+    ULONGLONG userModeTime() const
+    {
+        return m_userModeTime.QuadPart;
+    }
+
+    DWORD pageFileUsage() const
+    {
+        return m_pageFileUsage;
+    }
+
+    DWORD virtualSize() const
+    {
+        return m_virtualSize;
+    }
+
+    DWORD sessionId() const
+    {
+        DWORD ret{};
+        if (!ProcessIdToSessionId(m_pId, &ret))
+        {
+            throw std::system_error
+            {
+                static_cast<int>(GetLastError()),
+                std::system_category(),
+                "Unable to retrieve session ID from process with PID %lu" + m_pId
+            };
+        }
+        return ret;
+    }
+
+private:
+    void setProcessTimes()
+    {
+        FILETIME lpCreationTime{};
+        FILETIME lpExitTime{};
+        FILETIME lpKernelTime{};
+        FILETIME lpUserTime{};
+        if (GetProcessTimes(m_spProcess, &lpCreationTime, &lpExitTime, &lpKernelTime, &lpUserTime))
+        {
+            // Copy the kernel mode filetime high and low parts and convert it to seconds
+            m_kernelModeTime.LowPart = lpKernelTime.dwLowDateTime;
+            m_kernelModeTime.HighPart = lpKernelTime.dwHighDateTime;
+            m_kernelModeTime.QuadPart /= 10000000ULL;
+
+            // Copy the user mode filetime high and low parts and convert it to seconds
+            m_userModeTime.LowPart = lpUserTime.dwLowDateTime;
+            m_userModeTime.HighPart = lpUserTime.dwHighDateTime;
+            m_userModeTime.QuadPart /= 10000000ULL;
+        }
+        else
+        {
+            throw std::system_error
+            {
+                static_cast<int>(GetLastError()),
+                std::system_category(),
+                "Unable to retrieve kernel mode and user mode times from process with PID %lu" + m_pId
+            };          
+        }
+    }
+
+    void setProcessMemInfo()
+    {
+        PROCESS_MEMORY_COUNTERS pMemCounters;
+        // Get page file usage and virtual size
+        // Reference: https://stackoverflow.com/a/1986486
+        if (GetProcessMemoryInfo(m_spProcess, &pMemCounters, sizeof(pMemCounters)))
+        {
+            m_pageFileUsage = pMemCounters.PagefileUsage;
+            m_virtualSize   = pMemCounters.WorkingSetSize + pMemCounters.PagefileUsage;
+        }
+        else
+        {
+            throw std::system_error
+            {
+                static_cast<int>(GetLastError()),
+                std::system_category(),
+                "Unable to retrieve page file usage from process with PID %lu" + m_pId
+            };
+        }
+    }
+
+    int ntPath2Win32Path(char* ntpath, char** outbuf)
```

Comment:
maybe you can use a static map<device name, dos name> and fill it the first time a dospath is needed and if the device name is not found in the map re fill it (the might be a new volumne or disk in the system).